### PR TITLE
At/docs

### PR DESCRIPTION
**Pull request by thazhemadam**
Add a few docstrings, and new section for `Codec`s

### CODE REVIEW

**Comment by rkurchin** [comment]
Can you grab commits 3c7032 through 4581d1c from #87 (excluding 04a1b78, I don't think we need to tag this) and add them to this since that's probably not going to get merged and those definitely fit under this header?

Also, for my own educational purposes, can you send me the lines of git that you run to do that? 🥺

**Comment by rkurchin** [comment]
(All the other stuff here looks great to me, thanks for adding those missing docstrings!)

File: src/features/elementfeature.jl [hunk 1]
```diff
@@ -91,8 +105,20 @@ function (ed::OneHotOneCold)(
...
+"""
+    output_shape(efd::ElementFeatureDescriptor)
+
+Get the output-shape for an ElementFeatureDescriptor object using the logic assoicated with its
+Codec.
+"""
 output_shape(efd::ElementFeatureDescriptor) = output_shape(efd, efd.encoder_decoder)
 
+"""
+    output_shape(efd::ElementFeatureDescriptor, ed::OneHotOneCold)
```
Comment (rkurchin) [inline src/features/elementfeature.jl:117]: I'm not sure this actually needs to be a separate docstring?

Comment (thazhemadam) [reply src/features/elementfeature.jl:117]: fair enough.

File: test/runtests.jl [hunk 2]
```diff
@@ -4,6 +4,7 @@ using Test
 const testdir = dirname(@__FILE__)
 
 tests = [
+    "module_tests.jl"
```
[the remark this answers is not in the document]
Comment (thazhemadam) [reply test/runtests.jl:7]: oops, fixing it now.

**Pull request merged**